clearpath_config/platform/drivetrain.py

```diff
@@ -0,0 +1,305 @@
+# Software License Agreement (BSD)
+#
+# @author    Roni Kreinin <rkreinin@clearpathrobotics.com>
+# @copyright (c) 2025, Clearpath Robotics, Inc., All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# * Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# * Neither the name of Clearpath Robotics nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+from clearpath_config.common.types.config import BaseConfig
+from clearpath_config.common.types.platform import Platform
+from clearpath_config.common.utils.dictionary import flip_dict
+
+
+class DrivetrainConfig(BaseConfig):
+    DRIVETRAIN = 'drivetrain'
+    CONTROL = 'control'
+    WHEELS = 'wheels'
+    FRONT = 'front'
+    REAR = 'rear'
+
+    # Types
+    DIFF_FWD = 'diff_fwd'
+    DIFF_RWD = 'diff_rwd'
+    DIFF_4WD = 'diff_4wd'
+    OMNI_4WD = 'omni_4wd'
+
+    # Wheels
+    OUTDOOR = 'outdoor'
```

Comment:
Do we want this to be `OFFROAD='offroad'` instead in case we add rally-style wheels intended for roads/asphalt, as opposed to the deep-tread standard wheels?

Reply:
I think I would prefer to keep this `outdoor` to match our outdoor/indoor robot naming convention. If we need to add another wheel type we can do that later.